app/v1/models/response/retrieveTransaction/RetrieveTransactionsResponse.scala

```diff
@@ -16,17 +16,52 @@
 
 package v1.models.response.retrieveTransaction
 
-import play.api.libs.json.{JsPath, Json, OWrites, Reads}
+import cats.Functor
+import config.AppConfig
+import play.api.libs.json._
+import v1.controllers.requestParsers.validators.validations.PaymentIdValidation
+import v1.hateoas.{HateoasLinks, HateoasListLinksFactory}
+import v1.models.hateoas.{HateoasData, Link}
 
 case class RetrieveTransactionsResponse[I](transactions: Seq[I])
 
 
-object RetrieveTransactionsResponse {
+object RetrieveTransactionsResponse extends HateoasLinks {
+
   implicit def reads[I: Reads]: Reads[RetrieveTransactionsResponse[I]] =
-  implicitly(JsPath \ "transactions").read[Seq[I]].map(items => RetrieveTransactionsResponse(items.filterNot(_ == TransactionItem.empty)))
+    implicitly(JsPath \ "transactions").read[Seq[I]].map(items => RetrieveTransactionsResponse(items.filterNot(_ == TransactionItem.empty)))
+
+  implicit def writes[I: Writes]: OWrites[RetrieveTransactionsResponse[I]] =
+    Json.writes[RetrieveTransactionsResponse[I]]
+
+  implicit object LinksFactory extends HateoasListLinksFactory[RetrieveTransactionsResponse, TransactionItem, RetrieveTransactionsHateoasData] {
+
+    override def itemLinks(appConfig: AppConfig, data: RetrieveTransactionsHateoasData, item: TransactionItem): Seq[Link] = {
+      val id = item.id.getOrElse("")
+
+      val isPayment = PaymentIdValidation.validate(id) == Seq()
```

Comment:
For context it might be easier to reference this as `PaymentIdValidation.validate(id) == Nil`.  

At first glance, without knowing what the validation returns it looks odd having a random empty sequence floating about.